tests/test_ilastik/test_applets/dataSelection/testDataSelectionGui.py

```diff
@@ -91,3 +93,80 @@ def test_locked_scale_select_does_not_trigger_gui_and_informs_user(
         intercept_info_popup.assert_called_once()
     assert 1 == index_changed_mock.call_count
     assert 0 == mock_gui.handleScaleSelected.call_count
+
+
+@pytest.fixture(
+    params=[  # Cases "scales in this role - scales in other role"
+        "multi-single",
+        "multifewer-multimore",
+        "multimore-multifewer",
+    ]
+)
+def mismatching_model(request) -> Tuple[QStandardItemModel, List[int], List[int]]:
+    m = QStandardItemModel()  # mock DatasetDetailedInfoTableModel
+    m.isEmptyRow = lambda _row: False
+    scale_options = {  # Scale options of the currently viewed ("this") role
+        "multi-single": {"0": "100, 100, 10", "1": "50, 50, 10"},
+        "multifewer-multimore": {"0": "100, 100, 10", "1": "50, 50, 10"},
+        "multimore-multifewer": {"0": "100, 100, 10", "1": "50, 50, 10", "2": "25, 25, 10"},
+    }
+    enabled_option_indices = {  # Which of the options match across all roles
+        "multi-single": [0],  # Another role has a single image that matches the first scale
+        "multifewer-multimore": [0, 1],  # Another role has a multiscale; current one's scales all have a match
+        "multimore-multifewer": [0, 1],  # Another role has a multiscale; one of current's scales not present
+    }
+    disabled_option_indices = {  # Which of the options don't have a match in other roles
+        "multi-single": [1],
+        "multifewer-multimore": [],
+        "multimore-multifewer": [2],
+    }
+    m.get_scale_options = lambda _row: scale_options[request.param]
+    m.get_common_scale_option_indices = lambda _row: enabled_option_indices[request.param]
+    m.is_scale_locked = lambda _row: False
+    data_rows = [
+        ["image", "precomputed://http://localhost:8000", "", "z: 1, y: 10, x: 10", "100, 100, 10", ""],
+        [],  # Empty row because the custom setModel replaces the model's last row with an AddFileButton
+    ]
+    for i, row in enumerate(data_rows):
+        m.insertRow(i, [QStandardItem(d) for d in row])
+    return m, enabled_option_indices[request.param], disabled_option_indices[request.param]
+
+
+@pytest.fixture
+def dataset_table_with_mismatching_roles(
+    qtbot, mismatching_model
+) -> Tuple[DatasetDetailedInfoTableView, List[int], List[int]]:
+    model, enabled, disabled = mismatching_model
+    table_view = DatasetDetailedInfoTableView(None)
+    table_view.setModel(model)
+    # Make sure qtbot window is big enough to `paint` also the Scale column
+    table_view.resize(1200, 300)
+    return prepare_widget(qtbot, table_view), enabled, disabled
+
+
+def test_scale_select_disables_scale_options_not_available_in_other_roles(
+    dataset_table_with_mismatching_roles, mock_gui, intercept_info_popup
```

Comment:
Love them or hate them - fixtures. In any case, my LSP will never figure out the types of these things unless explicitly annotated (does yours?). Can't live without it anymore ;)

Reply:
The typing doesn't work for me either :( I tried replacing the fixtures with plain functions, but qtbot doesn't render the table when I do that. Any idea how to fix that..?